### PR TITLE
Gui

### PR DESCRIPTION
**Pull request by ThatMathsyBardGuy**
- Moved file_dialog.svg to the resources folder, which the program now accesses through an IMAGEDIR macro.
- Slight addition to gitignore to avoid pushing any shaders that we're using to test things
- Added a new compile_all.bat to the shaders folder which will compile all .frag and .vert files in that folder

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -53,4 +53,11 @@ compile_commands.json
...
 # Build Directory 
 Build/
+
+#Shaders other than the test ones
+*.frag
+!fs_test.frag
+*.vert
+!vs_test.vert
+
 .vscode/
```
Comment (Kney-Delach) [inline .gitignore:64]: Could you add a section for editor specific files and add the .vscode/ to it, this should have been added in a previous merge but somebody forgot (yes it was me)  😅

**Review by Kney-Delach — CHANGES_REQUESTED** [review]
This merge is ✨! I've left a comment that should be reviewed before merging!

File: .gitignore (hunk 1 above)
Comment (ThatMathsyBardGuy) [reply .gitignore:64]: > 
> 
> This merge is ✨! I've left a comment that should be reviewed before merging!

Hmm can I just make that change, push to gui, and have this pull request update automatically like it did when master changed?

Comment (ThatMathsyBardGuy) [reply .gitignore:64]: Done!

Comment (Kney-Delach) [reply .gitignore:64]: Yes, perfect.

**Review by Kney-Delach — APPROVED** [review]
Everything seems good, you can merge now!

**Comment by ThatMathsyBardGuy** [comment]
> 
> 
> Everything seems good, you can merge now!

Is there a way to get rid of the "Changes Requested" thing? It'll let me merge anyway but it feels dirty
image

**Pull request merged**